src/main/java/com/list/todo/security/CurrentUser.java

```diff
@@ -0,0 +1,12 @@
+package com.list.todo.security;
+
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import java.lang.annotation.*;
+
+@Target({ElementType.PARAMETER, ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@AuthenticationPrincipal
+public @interface CurrentUser {
```

Comment:
here you just replace `@AuthenticationPrincipal` with your own `CurerrentUser`
why would you want that?